SLING-7625 refactor of the Default Get Servlet bundle

JEBailey opened the pull request:
In general
Remove all uses of deprecated API's
Remove all methods that aren't being used
Re-organize GET servlets to be in one location
Rename objects to better represent their purpose
Reduce memory consumption by removing unnecessary Object creation

I can go into specific details on any specific change

Commit:
Removed deprecated api calls

JEBailey committed:
SLING-7625 refactoring of the code base

Remove all uses of deprecated API's
Remove all methods that aren't being used
Re-organize GET servlets to be in one location
Rename objects to better represent their purpose
Reduce memory consumption by removing unnecessary Object creation

kwin commented on src/main/java/org/apache/sling/servlets/get/impl/DefaultGetServlet.java:
```diff
@@ -188,31 +189,18 @@ protected void deactivate() {
-    private Servlet getDefaultRendererServlet(final String type) {
-        Servlet servlet = null;
+    private Renderer getDefaultRendererServlet(final String type) {
+    	Renderer servlet = null;
```
Renaming the variable here would make sense as well.

kwin commented on src/main/java/org/apache/sling/servlets/get/impl/DefaultGetServlet.java:
```diff
@@ -223,7 +211,7 @@ public void init() throws ServletException {
         // use the servlet for rendering StreamRendererServlet.EXT_RES as the
         // streamer servlet
-        Servlet streamerServlet = getDefaultRendererServlet(EXT_RES);
+        Renderer streamerServlet = getDefaultRendererServlet(EXT_RES);
```
Renaming the variable here would make sense as well.

kwin commented on src/main/java/org/apache/sling/servlets/get/impl/DefaultGetServlet.java:
```diff
@@ -291,7 +279,7 @@ protected void doGet(SlingHttpServletRequest request,
         }
 
-        Servlet rendererServlet;
+        Renderer rendererServlet;
```
Renaming the variable here would make sense as well.

JEBailey replied:
good point

JEBailey committed:
SLING-7625 Removed servlet reference in Default Get Object

based on feedback updated method names, variable names, and comments to
remove reference to Servlet while referring to a renderer.
